pkg/genevalogging/genevalogging.go

```diff
@@ -154,107 +153,26 @@ type genevaLogging struct {
-
-	cli    kubernetes.Interface
-	seccli securityclient.Interface
+	oc      *api.OpenShiftCluster
+	ensurer ensure.Interface
 }
 
-func New(log *logrus.Entry, e env.Interface, oc *api.OpenShiftCluster, cli kubernetes.Interface, seccli securityclient.Interface) GenevaLogging {
+func New(log *logrus.Entry, e env.Interface, oc *api.OpenShiftCluster, ensure ensure.Interface) GenevaLogging {
 	return &genevaLogging{
 		log: log,
 		env: e,
 
-		oc: oc,
-
-		cli:    cli,
-		seccli: seccli,
-	}
-}
-
-func (g *genevaLogging) ensureNamespace(ns string) error {
-	_, err := g.cli.CoreV1().Namespaces().Create(&v1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: ns,
-		},
-	})
-	if !errors.IsAlreadyExists(err) {
-		return err
-	}
-
-	return nil
-}
-
-func (g *genevaLogging) applyConfigMap(cm *v1.ConfigMap) error {
-	_, err := g.cli.CoreV1().ConfigMaps(cm.Namespace).Create(cm)
-	if !errors.IsAlreadyExists(err) {
-		return err
-	}
-
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		_cm, err := g.cli.CoreV1().ConfigMaps(cm.Namespace).Get(cm.Name, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-
-		cm.ResourceVersion = _cm.ResourceVersion
-		_, err = g.cli.CoreV1().ConfigMaps(cm.Namespace).Update(cm)
-		return err
-	})
-}
-
-func (g *genevaLogging) applySecret(s *v1.Secret) error {
-	_, err := g.cli.CoreV1().Secrets(s.Namespace).Create(s)
-	if !errors.IsAlreadyExists(err) {
-		return err
+		oc:      oc,
+		ensurer: ensure,
 	}
-
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		_s, err := g.cli.CoreV1().Secrets(s.Namespace).Get(s.Name, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-
-		s.ResourceVersion = _s.ResourceVersion
-		_, err = g.cli.CoreV1().Secrets(s.Namespace).Update(s)
-		return err
-	})
 }
 
-func (g *genevaLogging) applyServiceAccount(sa *v1.ServiceAccount) error {
-	_, err := g.cli.CoreV1().ServiceAccounts(sa.Namespace).Create(sa)
-	if !errors.IsAlreadyExists(err) {
-		return err
-	}
-
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		_sa, err := g.cli.CoreV1().ServiceAccounts(sa.Namespace).Get(sa.Name, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-
-		sa.ResourceVersion = _sa.ResourceVersion
-		_, err = g.cli.CoreV1().ServiceAccounts(sa.Namespace).Update(sa)
-		return err
-	})
+func (g *genevaLogging) fluentbitImage() string {
+	return fmt.Sprintf(fluentbitImageFormat, g.env.ACRName())
 }
 
-func (g *genevaLogging) applyDaemonSet(ds *appsv1.DaemonSet) error {
-	_, err := g.cli.AppsV1().DaemonSets(ds.Namespace).Create(ds)
-	if !errors.IsAlreadyExists(err) {
-		return err
-	}
-
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		_ds, err := g.cli.AppsV1().DaemonSets(ds.Namespace).Get(ds.Name, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-
-		ds.ResourceVersion = _ds.ResourceVersion
-		_, err = g.cli.AppsV1().DaemonSets(ds.Namespace).Update(ds)
-		return err
-	})
+func (g *genevaLogging) mdsdImage() string {
```

Comment:
this is in pkg/util/version, don't bring it back here (messed up rebase?)

Reply:
Yeah rebase picked up some vendor stuff and I had to bring this back from the diff in Geneva logging.  This was a couple of days ago. I can make the change.